src/WinRT.Runtime/ActivationFactory.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using WinRT.Interop;
+
+namespace WinRT
+{
+    internal unsafe sealed class DllModule
+    {
+        readonly string _fileName;
+        readonly IntPtr _moduleHandle;
+        readonly delegate* unmanaged[Stdcall]<IntPtr, IntPtr*, int> _GetActivationFactory;
+        readonly delegate* unmanaged[Stdcall]<int> _CanUnloadNow; // TODO: Eventually periodically call
+
+        static readonly string _currentModuleDirectory = AppContext.BaseDirectory;
+
+        static Dictionary<string, DllModule> _cache = new System.Collections.Generic.Dictionary<string, DllModule>(StringComparer.Ordinal);
+
+        public static bool TryLoad(string fileName, out DllModule module)
+        {
+            lock (_cache)
+            {
+                if (_cache.TryGetValue(fileName, out module))
+                {
+                    return true;
+                }
+                else if (TryCreate(fileName, out module))
+                {
+                    _cache[fileName] = module;
+                    return true;
+                }
+                return false;
+            }
+        }
+
+        private static unsafe bool TryCreate(string fileName, out DllModule module)

```

Comment:
Nit: redundant `unsafe` modifier, it's already on the type declaration. Same for the other method below.